src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -550,10 +550,21 @@ private Parameter BuildParameter(RequestParameter requestParameter)
             {
                 defaultValue = Constant.Default(type);
             }
+
+            CSharpType parameterType = TypeFactory.GetInputType(type);
```

Comment:
We are already messing with the content-type parameter in LowLevelRestClient, can we fix the type there as well?

Reply:
I think you can turn Parameter into a record and use `contentTypeParameter with { TYpe = ... }`